### PR TITLE
[bridge][L1][L2] Deposit&Withdraw testing scripts

### PR DESCRIPTION
**Pull request by GuillermoLunar7**
Jira: PTP-12560

### CODE REVIEW

File: solidity/README.md [hunk 1]
```diff
@@ -20,3 +29,13 @@ pipx install eth-brownie
 ```bash
 brownie compile --all
 ```
+
+
```
Comment (varunjain15) [inline solidity/README.md:33]: nit: we can add steps here to import account for l1_admin and the test acc that we are trying to mint too

File: solidity/scripts/check_balance.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    """
+
+    user_account = accounts.add(USER_PRIVATE_KEY)
+
+    from_user_account = {"from": user_account}
+    usdc = USDCToken.at(L1_TOKEN_ADDRESS)
+
+    tx = usdc.balanceOf(user_account, from_user_account)
+
+    print("new balance ", tx)
```
Comment (varunjain15) [inline solidity/scripts/check_balance.py]: we can return quantum / decimal_places converted balance for easier interpretation 
`bal * 10 ** (TOKEN_QUANTUMS * -1)`

**Review by varunjain15 — APPROVED** [review]
LGTM with minor comments (+ review from Amay/Sergey)
Can leverage most of this for setting up GH action as well for minting to L1

File: solidity/contracts/external/tokens/ERC20/USDC.sol [hunk 3]
```diff
@@ -1,6 +1,52 @@
...
+     * be displayed to a user as `5.000005` (`5000005 / 10 ** 6`).
+     *
+     * NOTE: This information is only used for _display_ purposes: it in
+     * no way affects any of the arithmetic of the contract, including
+     * {IERC20-balanceOf} and {IERC20-transfer}.
+     */
+    function decimals() public view virtual override returns (uint8) {
+        return 6;
+    }
+}
```
Comment (amayvs) [inline solidity/contracts/external/tokens/ERC20/USDC.sol]: Missing newline at end of file

File: solidity/README.md (hunk 1 above)
Comment (GuillermoLunar7) [reply solidity/README.md:33]: simplified this, in the code you can see that only the private key is needed, will also add the way to do it with that

File: solidity/scripts/check_balance.py (hunk 2 above)
Comment (GuillermoLunar7) [reply solidity/scripts/check_balance.py]: For sure!

File: cairo/scripts/track_l2_message.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+print(
+    tx_hash_from_message(
+        from_address="0x000000000000000000000000bc5bf8f315b9d1eebd8062cadf113a05c5abb43e",
+        to_address="0x00d256efe3d853cd5422f7dd6ddf4d8fb73a1c7ce61cce1c00142b4c156351bf",
+        selector="0x02d757788a8d8d6f21d1cd40bce38a8222d70654214e96ff95d8086e684fbee5",
+        nonce=3,
+        payload=[
+            "023a137ff67f2d70c5f87a86f5829af1d2bde86ae9efcedf050fe7a096cd6768",
+            "00000000000000000000000000000000000000000000000000038d7ea4c68000",
+            "0000000000000000000000000000000000000000000000000000000000000000",
```
Comment (amayvs) [inline cairo/scripts/track_l2_message.py]: Is this required to be 64 length?

File: solidity/.gitignore [hunk 5]
```diff
@@ -4,3 +4,4 @@ __pycache__
 .hypothesis/
 build/
 reports/
+dev/
```
Comment (amayvs) [inline solidity/.gitignore]: Missing newline at end of file

**Review by amayvs — APPROVED** [review]

File: cairo/scripts/deploy_bridge.py [hunk 6]
```diff
@@ -50,17 +47,20 @@ async def deploy():
...
+    # L2 Bridge deployed and L2 Token deployed.
+    set_l1_bridge_invoke = await bridge.functions['set_l1_bridge'].invoke(
         int_16(L1_BRIDGE_ADDRESS),
         max_fee=int(1e16),
     )
-    print("Waiting for set_l1_bridge tx to be accepted...", hex(tx.hash))
-
-    tx = await bridge.functions['set_l2_token'].invoke(
+    print("Waiting for set_l1_bridge tx to be accepted...", hex(set_l1_bridge_invoke.hash))
+    await set_l1_bridge_invoke.wait_for_acceptance(wait_for_accept=True)
```
Comment (sv) [inline cairo/scripts/deploy_bridge.py:57]: seems like we can drop wait_for_accept=True in most cases

File: solidity/contracts/external/tokens/ERC20/USDC.sol [hunk 7]
```diff
@@ -1,6 +1,52 @@
...
-import "./ERC20.sol";
+import { AccessControl } from "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/AccessControl.sol";
+import { ERC20 } from "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/token/ERC20/ERC20.sol";
+import { ERC20Burnable } from "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import { Pausable } from "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/security/Pausable.sol";
 
-contract USDCToken is ERC20 {}
\ No newline at end of file
+/// @custom:security-contact security@paradex.trade
+contract USDCToken is ERC20, ERC20Burnable, Pausable, AccessControl {
```
Comment (sv) [inline solidity/contracts/external/tokens/ERC20/USDC.sol:10]: Is this for testing?

**Review by sv — APPROVED** [review]

File: solidity/contracts/external/tokens/ERC20/USDC.sol (hunk 7 above)
Comment (GuillermoLunar7) [reply solidity/contracts/external/tokens/ERC20/USDC.sol:10]: Yes our new L1 fake token

File: cairo/scripts/track_l2_message.py (hunk 4 above)
Comment (GuillermoLunar7) [reply cairo/scripts/track_l2_message.py]: Actually no, it can be `0`

**Pull request merged**